pennylane/spin/spin_hamiltonian.py

```diff
@@ -298,3 +309,347 @@ def fermi_hubbard(
+                        + FermiWord({(0, j * spin + 1): "+", (1, j * spin + 1): "-"})
+                    )
+                )
+
+    else:
+        for edge in lattice.edges:
+            i, j, order = edge
+            for s in range(spin):
+                s1 = i * spin + s
+                s2 = j * spin + s
+                hopping_ham -= hopping[i][j] * (
+                    FermiWord({(0, s1): "+", (1, s2): "-"})
+                    + FermiWord({(0, s2): "+", (1, s1): "-"})
+                )
+
+            if intersite_int.shape == (neighbour_order,):
+                intersite_term += (
+                    intersite_int[order]
+                    * (
+                        FermiWord({(0, i * spin): "+", (1, i * spin): "-"})
+                        + FermiWord({(0, i * spin + 1): "+", (1, i * spin + 1): "-"})
+                    )
+                    * (
+                        FermiWord({(0, j * spin): "+", (1, j * spin): "-"})
+                        + FermiWord({(0, j * spin + 1): "+", (1, j * spin + 1): "-"})
+                    )
+                )
+            else:
+                intersite_term += (
+                    intersite_int[i][j]
+                    * (
+                        FermiWord({(0, i * spin): "+", (1, i * spin): "-"})
+                        + FermiWord({(0, i * spin + 1): "+", (1, i * spin + 1): "-"})
+                    )
+                    * (
+                        FermiWord({(0, j * spin): "+", (1, j * spin): "-"})
+                        + FermiWord({(0, j * spin + 1): "+", (1, j * spin + 1): "-"})
+                    )
+                )
+
+    coulomb_term = 0.0 * FermiWord({})
+    if isinstance(coulomb, (int, float, complex)):
+        coulomb = math.ones(lattice.n_sites) * coulomb
+
+    for i in range(lattice.n_sites):
+        up_spin = i * spin
+        down_spin = i * spin + 1
+        coulomb_term += coulomb[i] * FermiWord(
+            {(0, up_spin): "+", (1, up_spin): "-", (2, down_spin): "+", (3, down_spin): "-"}
+        )
+
+    hamiltonian = hopping_ham + coulomb_term + intersite_term
+
+    if mapping not in ["jordan_wigner", "parity", "bravyi_kitaev"]:
+        raise ValueError(
+            f"The '{mapping}' transformation is not available."
+            f"Please set mapping to 'jordan_wigner', 'parity', or 'bravyi_kitaev'"
+        )
+    qubit_ham = qml.qchem.qubit_observable(hamiltonian, mapping=mapping)
+
+    return qubit_ham.simplify()
+
+
+def haldane(
+    lattice,
+    n_cells,
+    hopping1=1.0,
+    hopping2=1.0,
+    phi=1.0,
+    boundary_condition=False,
+    mapping="jordan_wigner",
+):
+    r"""Generates the Haldane Hamiltonian on a lattice.
+
+    The Hamiltonian is represented as:
+
+    .. math::
+
+        \hat{H} = -t_1\sum_{<i,j>, \sigma}(c_{i\sigma}^{\dagger}c_{j\sigma}) -t_2 \sum_{<<i,j>>, \sigma}(e^{i\phi_{ij}}c_{i\sigma}^{\dagger}c_{j\sigma+e^{-i\phi_{ij}}c_{j\sigma}^{\dagger}c_{i\sigma})
```

Comment:
We should also specify that ``<<i,j>>`` mean next-nearest neighbours in the description.